addons/account/models/account_move_line.py

```diff
@@ -2328,12 +2328,17 @@ def _reconcile_plan(self, reconciliation_plan):
 
         :param reconciliation_plan: A list of reconciliation to perform.
         """
-        # Parameter allowing to disable the exchange journal entries on partials.
-        disable_partial_exchange_diff = bool(self.env['ir.config_parameter'].sudo().get_param('account.disable_partial_exchange_diff'))
-
         # ==== Prepare the reconciliation ====
         # Batch the amls all together to know what should be reconciled and when.
         plan_list, all_amls = self._optimize_reconciliation_plan(reconciliation_plan)
+        move_container = {'records': all_amls.move_id}
+        with self.move_id._check_balanced(move_container),\
+             self.move_id._sync_dynamic_lines(move_container):
+             self.__reconcile_plan(plan_list, all_amls)
+
+    def __reconcile_plan(self, plan_list, all_amls):
```

Comment:
it's confusing to have both `self` & `all_amls` here

Reply:
I just wanted to avoid some diff to rename the variables :shrug:

Reply:
@william-andre `_reconcile_plan_with_sync` ? :sob: